services/src/pro/datasets/postgres.rs

```diff
@@ -279,11 +279,16 @@ where
                 source: Box::new(e),
             })?;
 
-        let meta_data: StaticMetaData<
-            OgrSourceDataset,
-            VectorResultDescriptor,
-            VectorQueryRectangle,
-        > = serde_json::from_value(row.get(0))?;
+        let meta_data: MetaDataDefinition = row.get("meta_data");
+
+        let MetaDataDefinition::OgrMetaData(meta_data) = meta_data else {
+                return Err(geoengine_operators::error::Error::MetaData {
+                    source: Box::new(geoengine_operators::error::Error::InvalidType {
+                        expected: "OgrMetaData".to_string(),
```

Comment:
also here: can you use the `type_name` method?

Reply:
Maybe you can give me a hint how to do that since I don't want to construct an OgrMetaData enum variant for that.